resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/events/AutoBuildJob.java

```diff
@@ -14,22 +14,34 @@
  *******************************************************************************/
 package org.eclipse.core.internal.events;
 
-import org.eclipse.core.internal.resources.*;
+import org.eclipse.core.internal.resources.ICoreConstants;
+import org.eclipse.core.internal.resources.ResourceException;
+import org.eclipse.core.internal.resources.Workspace;
 import org.eclipse.core.internal.utils.Messages;
 import org.eclipse.core.internal.utils.Policy;
-import org.eclipse.core.resources.*;
-import org.eclipse.core.runtime.*;
-import org.eclipse.core.runtime.Preferences.PropertyChangeEvent;
+import org.eclipse.core.resources.IResourceChangeEvent;
+import org.eclipse.core.resources.IncrementalProjectBuilder;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.SubMonitor;
 import org.eclipse.core.runtime.jobs.ISchedulingRule;
 import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences.PreferenceChangeEvent;
+import org.eclipse.core.runtime.preferences.InstanceScope;
 import org.osgi.framework.Bundle;
 
 /**
  * The job for performing workspace auto-builds, and pre- and post- autobuild
  * notification.  This job is run whenever the workspace changes regardless
  * of whether autobuild is on or off.
  */
-class AutoBuildJob extends Job implements Preferences.IPropertyChangeListener {
+class AutoBuildJob extends Job implements IEclipsePreferences.IPreferenceChangeListener {
```

Comment:
I highly appreciate cleanup of that deprecated stuff. Thanks!
If you like you could do it even better if AutoBuildJob does not implement anything but creates an private listener instance. That way users of AutoBuildJob could not recognize if the implementation changes like now.

Reply:
I don't like the idea of having to create another instance just to hide a listener method in an internal class, I prefer the straightforward approach which in my opinion is easier to read.
Users of `AutoBuildJob` should not expect that the class is an instance of `IEclipsePreferences.IPreferenceChangeListener` or invoke `preferenceChange(PreferenceChangeEvent)` directly.